src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsBrowserError.java

```diff
@@ -0,0 +1,47 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.testng.annotations.Test;
+
+/**
+ * @ownership AdEng
+ */
+public class TestAdsBrowserError extends TemplateNoFirstLoad {
+
+  @Test(groups = "AdsScrollHandlerBrowserErrorMercury")
+  public void adsScrollHandlerBrowserErrorMercury() throws InterruptedException {
+    String testPage = urlBuilder.getUrlForPath("adtest", "Wikia_Ad_Testing");
+    testPage = urlBuilder.appendQueryStringToURL(testPage, "scrollhandler=1");
+    AdsBaseObject adsBaseObject = new AdsBaseObject(driver);
+    adsBaseObject.getUrl(testPage);
+
+    executeScript(addErrorListenerScript());
+    executeScript(scroll());
+
+    adsBaseObject.waitForJavaScriptTruthy("window.scrollY != 0");
+    Assertion.assertEquals(executeScript(getErrors()), "");
+  }
+
+  private String addErrorListenerScript() {
```

Comment:
@justnpT :+1: 
> I was thinkig about storing Javascripts in .js files and reading them in the java methods, then pushing the javascripts to execute them in the webdriven browser

I was thinking about it as well, but it will require much more work which I can't spend on this ticket. We need to read JS file with a lot of methods as a String and parse it _wisely_ and pass it to Selenium `executeScript`. If you would implement such mechanism in our repo I would be happy to use it and to move my JS logic into separate JS file.